### PR TITLE
Bump up kopia v0.13

### PR DESCRIPTION
**Pull request by Lyndon-Li**
Bump up kopia to v0.13

Fix issue #5999

### CODE REVIEW

**Comment by Lyndon-Li** [comment]
We face [this problem](https://github.com/kubernetes-sigs/controller-tools/issues/613) of controller-gen after bumping up and changing go.mod, as a result, the `make update` always crashes. So we need to bump up controller-gen to v0.12.0.

File: pkg/repository/udmrepo/repo.go [hunk 1]
```diff
@@ -68,6 +68,7 @@ type ObjectWriteOptions struct {
 	Prefix      ID     // A prefix of the name used to save the object
 	AccessMode  int    // OBJECT_DATA_ACCESS_*
 	BackupMode  int    // OBJECT_DATA_BACKUP_*
+	AsyncWrites int    // Num of async writes for the object, 0 means no async write
```
Comment (blackpiglet) [inline pkg/repository/udmrepo/repo.go:71]: Looks like there is no place to assign value to the `AsyncWrites` parameter.

File: pkg/uploader/kopia/shim.go [hunk 2]
```diff
@@ -208,11 +208,22 @@ func (sr *shimRepository) DeleteManifest(ctx context.Context, id manifest.ID) er
...
 // Flush all the unifited repository data
 func (sr *shimRepository) Flush(ctx context.Context) error {
 	return sr.udmRepo.Flush(ctx)
 }
 
+func (sr *shimRepository) ConcatenateObjects(ctx context.Context, objectIDs []object.ID) (object.ID, error) {
+	return object.ID{}, errors.New("ConcatenateObjects is not supported")
+}
+
+func (sr *shimRepository) OnSuccessfulFlush(callback repo.RepositoryWriterCallback) {
```
Comment (blackpiglet) [inline pkg/uploader/kopia/shim.go:224]: Just curious about why not `OnSuccessfulFlush` function return value handling is different from other newly added functions.

**Comment by blackpiglet** [comment]
Could you give more information on why the uber zap log library is used instead in `pkg/util/logging/kopia_log.go`

File: pkg/repository/udmrepo/repo.go (hunk 1 above)
Comment (Lyndon-Li) [reply pkg/repository/udmrepo/repo.go:71]: Fixed in `(sr *shimRepository) NewObjectWriter`, it is an optional value, so right now, we don't need to set a value, just convey the value from shim to unified repo.

**Comment by Lyndon-Li** [comment]
> Could you give more information on why the uber zap log library is used instead in `pkg/util/logging/kopia_log.go`?

`pkg/util/logging/kopia_log.go` is still used. zap log is the logger used by Kopia all the time. 
In the previous release, Kopia has a wrapper of the zap log, so Velero doesn't need to hook zap log in order to redirect Kopia log to Velero.
In the current release, zap log is used directly by Kopia, so in Velero code we have to hook zap log for the same purpose.

File: pkg/uploader/kopia/shim.go (hunk 2 above)
Comment (Lyndon-Li) [reply pkg/uploader/kopia/shim.go:225]: The prototype of these functions are defined by Kopia, specifically, they implements Kopia' Repository interface. In the interface, `OnSuccessfulFlush` has no return value.
On the other hand, this function is a callback under the Flush operation, while this callback is optional, if it is not to be handled, just leave it as empty.

**Review by blackpiglet — APPROVED** [review]

**Review by qiuming-best — APPROVED** [review]

**Pull request merged**